octoprint_automation_scripts/__init__.py

```diff
@@ -133,36 +157,54 @@ def execute_script(self, scriptname):
 
             # Ensure that any commands sent to the printer are actually executed
             # *before* cleaning up the script object.
-            if result.get('wait_for_buffer'):
+            if result['wait_for_buffer']:
                 self.g.write("M400", resp_needed=True)
             self.so = None
 
             # Store script's settings.
-            if result.get('storage') is not None:
+            if result['success'] and result['storage'] is not None:
                 for key, val in result['storage'].iteritems():
-                    self._settings.set([scriptname, key], str(val))
+                    self._settings.set([script_id, key], str(val))
 
         except Exception as e:
             self._logger.exception('Script was forcibly exited: ' + str(e))
+            payload = {'id': script_id,
+                       'title': self.script_titles[script_id],
+                       'error': str(e)}
+            eventManager().fire(Events.AUTOMATION_SCRIPT_ERROR, payload)
             self.relinquish_control(wait=False)
             return
 
         self.relinquish_control()
 
+        if result['success']:
```

Comment:
If anything in there raises the `except` block starting on line 169 will be executed which includes a return statement.